### PR TITLE
Encoding chapter

### PR DESCRIPTION
**Pull request by giacomocavalieri**

### CODE REVIEW

File: packages/docs/docs/docs/model/03-encode.md [hunk 1]
```diff
@@ -1 +1,152 @@
...
+The great advantage provided by the result type is that it _forces_ you to deal
+with the error case, focusing your attention not only on the happy path but also
+on what could go wrong.
+
+To get a value out of a `Result` one must first check wether it is successful or
+not:
+
+```ts showLineNumbers
+function printResult(res: Result<number, string>) {
+    // highlight-start
```
Comment (edobrb) [inline packages/docs/docs/docs/model/03-encode.md:87]: highlight-start and highlight-end should do something visually?

File: packages/docs/docs/docs/model/03-encode.md [hunk 2]
```diff
@@ -1 +1,152 @@
...
+inferred type, it is not possible to statically check that values of the
+inferred type respect those invariants: it is the encoder's job to make sure an
+invalid value never gets encoded.
+
+In the previous example, `NonNegativeNumber` is inferred as a simple
+Typescript `number`, so the `encode` method has to check that it is a
+non-negative number. When given a negative number the encoding process fails
+with an error describing what went wrong.
+
+## A small introduction on the `Result` type
```
Comment (edobrb) [inline packages/docs/docs/docs/model/03-encode.md:40]: Not sure if we can leave `Result` here or if it deserves it's own chapter.

File: packages/docs/docs/docs/model/03-encode.md [hunk 3]
```diff
@@ -1 +1,152 @@
...
+
+const value = { name: "Giacomo", secret: "..." }
+const json = user.encode(value) // -> { name: "Giacomo", secret: "..." }
+shareJsonValue(json) // Uh oh, we ended up sharing the secret value!
+```
+
+In this example every single field ends up encoded in the final object, so we
+end up sharing the user's secret.
+
+One wai to fix this problem is to remember to remove the sensitive data from
```
Comment (edobrb) [inline packages/docs/docs/docs/model/03-encode.md:132]: Maybe mention that sensitive data are always encoded to null with `hide` strategy

File: packages/docs/docs/docs/model/03-encode.md (hunk 1 above)
Comment (giacomocavalieri) [reply packages/docs/docs/docs/model/03-encode.md:87]: Yes, they highlight the lines between `start` and `end` and hide those two comments

File: packages/docs/docs/docs/model/03-encode.md (hunk 2 above)
Comment (giacomocavalieri) [reply packages/docs/docs/docs/model/03-encode.md:40]: I could also move it to its own chapter and link to it, tell me what you prefer

Comment (edobrb) [reply packages/docs/docs/docs/model/03-encode.md:40]: Leave it there for now 👍

**Pull request merged**